website/identifiers/clients/crossref.py

```diff
@@ -138,6 +138,23 @@ def build_posted_content(self, preprint, element, include_relation):
         ]
         posted_content.append(element.doi_data(*doi_data))
 
+        preprint_versions = preprint.get_preprint_versions()
+        for preprint_version in preprint_versions:
+            preprint_identifier = preprint_version.identifiers.first().value
+            if doi != preprint_identifier:
```

Comment:
I don't think the current version has `identifier` saved yet when the CrossRef request is made.

Instead, we should exclude the current version first, something like

```python
if preprint_version == preprint:
    continue
```